Apps/FlightSoftware/Watchdog/src/i2c_sensors.c

```diff
@@ -0,0 +1,575 @@
+                keepSpinning = I2C_Sensors__gaugeTempLsb();
+                break;
+
+            case GRS__GAUGE_TEMP_MSB:
+                keepSpinning = I2C_Sensors__gaugeTempMsb();
+                break;
+
+            case GRS__DONE:
+                keepSpinning = FALSE;
+                break;
+
+            default:
+                break;
+        }
+    }
+}
+
+//###########################################################
+// Private function definitions
+//###########################################################
+
+BOOL I2C_Sensors__readRegNonBlocking(uint8_t devAddr,
+                                     uint8_t regAddr,
+                                     uint8_t nackMaskBit,
+                                     GaugeReadingState nextState,
+                                     void* output,
+                                     BOOL* done,
+                                     BOOL* gotOutput)
+{
+    static I2C__TransactionStatus tStatus = { 0 };
+
+    I2C__Status stat = I2C__getTransactionStatus(&tStatus);
+
+    BOOL startNewRead = FALSE;
+    BOOL continueSpinning = FALSE;
+
+    if (I2C__STATUS__ERROR__NO_TRANSACTION == stat) {
+        startNewRead = TRUE;
+    } else if (I2C__STATUS__SUCCESS == stat) {
+        BOOL statusForRequestedTransaction = FALSE;
+        BOOL prevReadDone = FALSE;
+
+        if ((tStatus.devAddr == devAddr) && (tStatus.regAddr == regAddr)) {
+            // This transaction status is for the currently requested read
+            statusForRequestedTransaction = TRUE;
+        }
+        
+        if (I2C__TRANSACTION__DONE_SUCCESS == tStatus.state) {
+            // We're now done with this read (and it was successful)
+            if (statusForRequestedTransaction) {
+                internals.gState = nextState;
+                *((uint8_t*) output) = tStatus.data;
+                *done = TRUE;
+                *gotOutput = TRUE;
+                continueSpinning = TRUE;
+            }
+
+            prevReadDone = TRUE;
+        } else if (I2C__TRANSACTION__DONE_ERROR_NACK == tStatus.state) {
+            // We're now done with this read (and it failed)
+            if (statusForRequestedTransaction) {
+                internals.readings.nackMask |= nackMaskBit;
+                internals.gState = nextState;
+                *done = TRUE;
+                *gotOutput = FALSE;
+                continueSpinning = TRUE;
+            }
+
+            prevReadDone = TRUE;
+        }
+
+        startNewRead = (!statusForRequestedTransaction && prevReadDone);
+    }
+
+    if (startNewRead) {
+        // If the NACK mask bit for this register is already set, just skip
+        // this register
+        if (internals.readings.nackMask & nackMaskBit != 0) {
+            internals.gState = nextState;
+            *done = TRUE;
+            *gotOutput = FALSE;
+            return TRUE;
+        }
+
+        stat = I2C__read(devAddr, regAddr);
+
+        assert(I2C__STATUS__SUCCESS == stat);
```

Comment:
For some reason, CCS tells me that "Function _assert could not be resolved" when those lines are uncommented. I tried a few suggested solutions I found online (including assert.h, adding --c11 flag to the compiler, using ASSERT instead of assert) but none of those helped at all

Reply:
@SchwartzCode Could it be [this](https://e2e.ti.com/support/tools/code-composer-studio-group/ccs/f/code-composer-studio-forum/835119/compiler-drv8889-q1evm-error-function-__bis_sr_register-could-not-be-resolved-function-_nop-could-not-be-resolved/3094211#3094211)? (Namely that the error isn't from the compiler, but is instead from the static analyzer?

Reply:
Looks like that is indeed the case! Resolved and asserts are now uncommented